tests/Requests/PayloadTest.php

```diff
@@ -110,4 +110,116 @@ public function it_translates_non_requests_to_an_empty_string()
+
+        $this->assertEquals($expected, (string) new Payload($request));
+    }
+
+    /**
+     * @test
+     */
+    public function it_stringifies_a_payload_with_the_好_character_to_use_the_escaped_string()
+    {
+        $request = (new GuzzleRequest('GET', 'https://localhost', [], '好'))
+            ->withHeader('X-SIGNED-ID', '303103f5-3dca-4704-96ad-860717769ec9')
+            ->withHeader('X-SIGNED-TIMESTAMP', '2018-04-06 20:34:47');
+
+        $expected = '{"id":"303103f5-3dca-4704-96ad-860717769ec9","method":"GET","timestamp":"2018-04-06 20:34:47","uri":"https://localhost","content":"好"}';
+
+        $this->assertEquals($expected, (string) new Payload($request));
+    }
+
+    /**
+     * @test
+     */
+    public function it_stringifies_a_json_payload_to_a_string()
+    {
+        $request = (new GuzzleRequest('GET', 'https://localhost', [], json_encode(['test' => 'test'])))
+            ->withHeader('X-SIGNED-ID', '303103f5-3dca-4704-96ad-860717769ec9')
+            ->withHeader('X-SIGNED-TIMESTAMP', '2018-04-06 20:34:47');
+
+        $expected = '{"id":"303103f5-3dca-4704-96ad-860717769ec9","method":"GET","timestamp":"2018-04-06 20:34:47","uri":"https://localhost","content":"{\"test\":\"test\"}"}';
+
+        $this->assertEquals($expected, (string) new Payload($request));
+    }
+
+    /**
+     * @test
+     */
+    public function it_stringifies_a_json_payload_with_the_ã_character_to_a_string()
+    {
+        $request = (new GuzzleRequest('GET', 'https://localhost', [], json_encode(['ã' => 'ã'], JSON_UNESCAPED_UNICODE)))
+            ->withHeader('X-SIGNED-ID', '303103f5-3dca-4704-96ad-860717769ec9')
+            ->withHeader('X-SIGNED-TIMESTAMP', '2018-04-06 20:34:47');
+
+        $expected = '{"id":"303103f5-3dca-4704-96ad-860717769ec9","method":"GET","timestamp":"2018-04-06 20:34:47","uri":"https://localhost","content":"{\"ã\":\"ã\"}"}';
+
+        $this->assertEquals($expected, (string) new Payload($request));
+    }
+
+    /**
+     * @test
+     */
+    public function it_stringifies_a_json_payload_with_the_好_character_to_a_string()
+    {
+        $request = (new GuzzleRequest('GET', 'https://localhost', [], json_encode(['好' => '好'], JSON_UNESCAPED_UNICODE)))
+            ->withHeader('X-SIGNED-ID', '303103f5-3dca-4704-96ad-860717769ec9')
+            ->withHeader('X-SIGNED-TIMESTAMP', '2018-04-06 20:34:47');
+
+        $expected = '{"id":"303103f5-3dca-4704-96ad-860717769ec9","method":"GET","timestamp":"2018-04-06 20:34:47","uri":"https://localhost","content":"{\"好\":\"好\"}"}';
+
+        $this->assertEquals($expected, (string) new Payload($request));
+    }
+
+    /**
+     * @test
+     */
+    public function it_stringifies_a_json_payload_with_a_url_to_a_string()
+    {
+        $request = (new GuzzleRequest('GET', 'https://localhost', [], json_encode(['url' => 'https://google.com'], JSON_UNESCAPED_UNICODE)))
```

Comment:
I think this is going to have to be `json_encode(['url' => 'https://google.com'], JSON_UNESCAPED_UNICODE|JSON_UNESCAPED_SLASHES)`. 

`JSON.stringify` doesn't escape the '/' in the URL.

Reply:
I believe we've made changes to account for this.